models/models.js

```diff
@@ -11,11 +11,41 @@ exports.fetchArticles = async (query) => {
 
   let { sort_by, order, topic } = query;
 
-  const [key] = Object.keys(query);
 
-  if (key && !validQueries.includes(key)) {
-    return Promise.reject({ status: 400, msg: `Invalid Input` });
-  }
+  //PR FEEDBACK -> cannot garuntee order 
+  //const [test] = Object.keys(query);
+  // if (test && !validQueries.includes(test)) {
+  //   return Promise.reject({ status: 400, msg: `Invalid Input` });
+  // }
+
+ 
+  //key should be checked for validity
+  //goood === {sort_by: created_at}
+  // bad === {sort: created_at}
+
+
+  //v0.3
+  const queryKeys = Object.keys(query);
+  queryKeys.forEach((query) => {
+    console.log(query)
+    if (!validQueries.includes(query)) {
+      console.log(query)
+      return Promise.reject({ status: 400, msg: "Invalid Input" });
```

Comment:
this error is not being throw , still receiving 200 ok , when console .log the error, it is showing an second promise rejection further down. unsure on why